CHANGELOG.md

```diff
@@ -1,5 +1,29 @@
 # MobX-React Changelog
 
+### 6.0.0
+
+**Breaking changes**
+
+* The minimum support version of React is 16.8.0
+* Killed the possibility to directly pass store names to `observer`. Always use `inject` instead. (This was deprecate for a long time already). `observer(["a", "b"], component)` should now be written as `inject("a", "b")(component)`.
+* `observer` components no longer automatically recover from errors (to prevent potential memory leaks). Instead, this is the responsibility of error boundaries.
+* `inject` now supports ref forwarding. As such, the `.wrappedInstance` property has been removed since refs can be used instead. (Fixes [#616](https://github.com/mobxjs/mobx-react/issues/616) (See also [#619](https://github.com/mobxjs/mobx-react/pull/619) by [42shadow42](https://github.com/42shadow42))
+* Changing the set of stores in `Provider` is no longer supported and while throw a hard error (this was a warning before), as the model of `Provider` / `inject` has always been to inject final values into the tree. (That is, fixed references, injected objects themselves can be stateful without problem). If you want to dynamically swap what is provided into the tree, use `React.createContext` instead of `Provider` / `inject`. The  suppressChangedStoreWarning` flag for `Provider` has been dropped.
+* The third argument of custom `storesToProps` functions passed to `inject` is no longer available.
+* `<Observer>` no longer supports the deprecated `inject` property.
+* Defining `shouldComponentUpdate` on `observer` compon
+* `propTypes` is no longer exposed, use `PropTypes` instead
+* `disposeOnUnmount` now only supports direct subclasses of `React.Component` and `React.PureComponent`. This prevents several unreliable edge cases that silently leaked memory before. Either only extend React.(Pure)Component when using `disposeOnUnmount`, or manually clean up stuff in `componentWillUnmount`.
```

Comment:
@xaviergonz how you feel about this one?

Reply:
fine by me. will it still work when arrow functions are used to declare lifecycles though?

Reply:
IIRC yes